src/OpenTelemetry.Instrumentation.StackExchangeRedis/TracerProviderBuilderExtensions.cs

```diff
@@ -119,8 +156,15 @@ public static class TracerProviderBuilderExtensions
             {
                 var instrumentation = sp.GetRequiredService<StackExchangeRedisInstrumentation>();
 
-                connection ??= sp.GetService<IConnectionMultiplexer>();
-
+#if NET8_0_OR_GREATER
+                connection ??= serviceKey == null
+                    ? sp.GetService<IConnectionMultiplexer>()
+                    : sp.GetKeyedService<IConnectionMultiplexer>(serviceKey);
+#else
+                connection ??= serviceKey == null
+                    ? sp.GetService<IConnectionMultiplexer>()
+                    : throw new InvalidOperationException("Redis instrumentation with service key is supported in .NET 8.0 and above.");
+#endif
```

Comment:
Based on https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/1451#issuecomment-1834172834. There is no need to conditional compilation. It is indirect reference by `"OpenTelemetry.Api.ProviderBuilderExtensions v1.8.1` 

```suggestion
                connection ??= serviceKey == null
                    ? sp.GetService<IConnectionMultiplexer>()
                    : sp.GetKeyedService<IConnectionMultiplexer>(serviceKey);
```